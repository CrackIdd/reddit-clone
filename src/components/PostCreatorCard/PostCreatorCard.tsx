/* eslint-disable jsx-a11y/img-redundant-alt */
import CSSModules from "react-css-modules";
import Card from "../Card/Card";
import styles from "./PostCreatorCard.module.css";
import { AiOutlinePicture } from "react-icons/ai";
import { BsLink45Deg } from "react-icons/bs";
import { isUserSignedIn } from "../../firebase";
import { useRef } from "react";
import { useNavigate } from "react-router-dom";
import { useAppSelector } from "../../hooks/hooks";
import { selectCommunityData } from "../../features/subreddit/subredditSlice";

const PostCreatorCard: React.FC = () => {
  const inputRef = useRef<HTMLInputElement | null>(null);

  const navigate = useNavigate();

  const { name } = useAppSelector(selectCommunityData);

  function inputFocus() {
    navigate(`/r/${name}/submit`);
  }
  return (
    <>
      {isUserSignedIn() && (
        <Card>
          <div styleName="post-creator-card">
            <div
              styleName="post-creator-card__picture"
              onClick={() => navigate("/user")}
            >
              <img
                styleName="post-creator-card__profile-icon"
                src="https://styles.redditmedia.com/t5_g2km0/styles/profileIcon_snood9e1c00c-b1c6-46a9-a231-a6fcd78cdd16-headshot.png?width=256&height=256&frame=1&crop=256:256,smart&s=7bd48be150588ab7dc3dc5c73be4d0dcddeeae8d"
                alt="default reddit profile"
              />
            </div>
            <input
              ref={inputRef}
              styleName="post-creator-card__input"
              placeholder="Create Post"
              onFocus={inputFocus}
            />
            <a styleName="post-creator-card__icon-link">
              <AiOutlinePicture styleName="post-creator-card__icon" />
            </a>
            <a styleName="post-creator-card__icon-link">
              <BsLink45Deg styleName="post-creator-card__icon" />
            </a>
          </div>
        </Card>
      )}
    </>
  );
};

export default CSSModules(PostCreatorCard, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
