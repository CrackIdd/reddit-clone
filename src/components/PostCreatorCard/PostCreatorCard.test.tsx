import { queryByPlaceholderText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import PostCreatorCard from "./PostCreatorCard";

describe("PostCreatorCard", () => {
  test("snapshot", () => {
    const { asFragment } = render(<PostCreatorCard />);

    expect(asFragment()).toMatchSnapshot();
  });
});
