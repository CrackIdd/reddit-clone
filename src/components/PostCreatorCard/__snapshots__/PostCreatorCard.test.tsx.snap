// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCreatorCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="post-creator-card"
    >
      <div
        class="post-creator-card__picture"
      >
        <img
          alt="default reddit profile"
          class="post-creator-card__profile-icon"
          src="https://styles.redditmedia.com/t5_g2km0/styles/profileIcon_snood9e1c00c-b1c6-46a9-a231-a6fcd78cdd16-headshot.png?width=256&height=256&frame=1&crop=256:256,smart&s=7bd48be150588ab7dc3dc5c73be4d0dcddeeae8d"
        />
      </div>
      <input
        class="post-creator-card__input"
        placeholder="Create Post"
      />
      <a
        class="post-creator-card__icon-link"
      >
        <svg
          class="post-creator-card__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 632H136v-39.9l138.5-164.3 150.1 178L658.1 489 888 761.6V792zm0-129.8L664.2 396.8c-3.2-3.8-9-3.8-12.2 0L424.6 666.4l-144-170.7c-3.2-3.8-9-3.8-12.2 0L136 652.7V232h752v430.2zM304 456a88 88 0 1 0 0-176 88 88 0 0 0 0 176zm0-116c15.5 0 28 12.5 28 28s-12.5 28-28 28-28-12.5-28-28 12.5-28 28-28z"
          />
        </svg>
      </a>
      <a
        class="post-creator-card__icon-link"
      >
        <svg
          class="post-creator-card__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.715 6.542 3.343 7.914a3 3 0 1 0 4.243 4.243l1.828-1.829A3 3 0 0 0 8.586 5.5L8 6.086a1.002 1.002 0 0 0-.154.199 2 2 0 0 1 .861 3.337L6.88 11.45a2 2 0 1 1-2.83-2.83l.793-.792a4.018 4.018 0 0 1-.128-1.287z"
          />
          <path
            d="M6.586 4.672A3 3 0 0 0 7.414 9.5l.775-.776a2 2 0 0 1-.896-3.346L9.12 3.55a2 2 0 1 1 2.83 2.83l-.793.792c.112.42.155.855.128 1.287l1.372-1.372a3 3 0 1 0-4.243-4.243L6.586 4.672z"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
